### ISSUE TITLE
Under certain circumstances, node positions are set to (0,0)

### ISSUE THREAD

**Issue by adaburrows** [lightningrodlabs/rea-playspace#174]
In both #125 and #146 (this ticket supplants both of those issues), node positions are set to (0,0) this is due to our global variable which doesn't track the full set of [node id, position] combinations for all nodes.

In the case of #146, there is a drag event fired where the new position of the node is set to the null value of the global because the code that sets the position never checks to see if the position is valid. There may possibly be another bug hidden in that, but tracking the positions per node and checking to make sure it's not null should be a good start.

This depends on issue #141.

**Comment by fosterlynn** [comment]
In case this helps, I think this experience is this issue.

I had several RS's and a few processes in a plan.  I clicked on one RS to move it, it went to (0,0), lost the event from it to a process.  Couldn't move the RS or anything else. Reloaded.  Clicked on another RS, the event on that one moved to the first one.  Still couldn't move any objects, exiting and coming back did not help.

**Comment by fosterlynn** [comment]
Update: I tried my other existing plans, and they all work fine.  I started a new plan, it also works fine.  So there is a fairly easy workaround.
